MiniApp/Classes/MiniAppLister.swift

```diff
@@ -1,2 +1,54 @@
 internal class MiniAppLister {
+    var environment: Environment
+    var miniAppClient: MiniAppClient
+
+    init(environment: Environment) {
+        self.environment = environment
+        self.miniAppClient = MiniAppClient()
+    }
+
+    func fetchList(completionHandler: @escaping (Result<[MiniAppInfo], Error>) -> Void) {
+        guard let url = environment.listingUrl else {
+            return completionHandler(.failure(self.invalidURLError()))
+        }
+
+        self.miniAppClient.requestFromServer(request: listingURLRequest(url: url)) { (result) in
+            switch result {
+            case .success(let responseData):
+                guard let decodeResponse = self.decodeListingResponse(with: responseData.data) as? [MiniAppInfo] else {
+                    return completionHandler(.failure(self.invalidResponseData()))
+                }
+                return completionHandler(.success(decodeResponse))
+            case .failure(let error):
+                return completionHandler(.failure(error))
+            }
+        }
+    }
+
+    func listingURLRequest(url: URL) -> URLRequest {
+        var urlRequest = URLRequest(url: url)
+        urlRequest.setAuthorizationHeader(environment: environment)
```

Comment:
Is it possible to set this in the `MiniAppClient` so you don't have to add it to every API request?

Reply:
We are preparing the URLRequest in MiniAppLister with listingURL, Headers and Query parameters. Moving just this wouldn't be odd ?
Even moving to MiniAppClient will set the authorization for every API request.

Reply:
I think you can set it on the session https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1411532-httpadditionalheaders

Reply:
We should not be modifying the headers in URLSession object as per the documentation.
"An URLSession object is designed to handle various aspects of the HTTP protocol for you. As a result, you should not modify the following headers:
* Authorization"